html/user/am_set_info.php

```diff
@@ -111,7 +112,24 @@ function success($x) {
 $send_email = BoincDb::escape_string($send_email);
 $show_hosts = BoincDb::escape_string($show_hosts);
 $venue = BoincDb::escape_string($venue);
-if ($email_addr) {
+$send_changed_email_to_user = false;
+$email_addr = strtolower($email_addr);
+if ($email_addr && $email_addr != $user->email_addr) {
+    $tmpuser = BoincUser::lookup_email_addr($email_addr);
+    if ($tmpuser) {
+        xml_error(ERR_BAD_EMAIL_ADDR, "There's already an account with that email address.");
+    }
+
+    //check if the email address is included in previous_email_addr window. 
+    //
+    $tmpuser = BoincUser::lookup_prev_email_addr($email_addr);
+    if ($tmpuser) {
+        xml_error(ERR_BAD_EMAIL_ADDR, "Email address is already in use");
```

Comment:
This and the above should probably be `ERR_DB_NOT_UNIQUE` to match creating accounts.